scripts/spack/configs/config.yaml

```
@@ -33,48 +42,163 @@ config:
 
   # Temporary locations Spack can try to use for builds.
   #
-  # Spack will use the first one it finds that exists and is writable.
-  # You can use $tempdir to refer to the system default temp directory
-  # (as returned by tempfile.gettempdir()).
+  # Recommended options are given below.
+  #
+  # Builds can be faster in temporary directories on some (e.g., HPC) systems.
+  # Specifying `$tempdir` will ensure use of the default temporary directory
+  # (i.e., ``$TMP` or ``$TMPDIR``).
+  #
+  # Another option that prevents conflicts and potential permission issues is
+  # to specify `~/.spack/stage`, which ensures each user builds in their home
+  # directory.
   #
-  # A value of $spack/var/spack/stage indicates that Spack should run
-  # builds directly inside its install directory without staging them in
-  # temporary space.
+  # A more traditional path uses the value of `$spack/var/spack/stage`, which
+  # builds directly inside Spack's instance without staging them in a
+  # temporary space.  Problems with specifying a path inside a Spack instance
+  # are that it precludes its use as a system package and its ability to be
+  # pip installable.
   #
-  # The build stage can be purged with `spack purge --stage`.
+  # In any case, if the username is not already in the path, Spack will append
+  # the value of `$user` in an attempt to avoid potential conflicts between
+  # users in shared temporary spaces.
+  #
+  # The build stage can be purged with `spack clean --stage` and
+  # `spack clean -a`, so it is important that the specified directory uniquely
+  # identifies Spack staging to avoid accidentally wiping out non-Spack work.
   build_stage:
     # skipping tempdir b/c running mpi tests fails with local fs
-    # - $tempdir
+    # - $tempdir/$user/spack-stage
     - $spack/../builds
+    # - ~/.spack/stage
+    # - $spack/var/spack/stage
 
+  # Directory in which to run tests and store test results.
+  # Tests will be stored in directories named by date/time and package
+  # name/hash.
+  test_stage: ~/.spack/test
 
-  # Cache directory already downloaded source tarballs and archived
-  # repositories. This can be purged with `spack purge --downloads`.
+  # Cache directory for already downloaded source tarballs and archived
+  # repositories. This can be purged with `spack clean --downloads`.
   source_cache: $spack/var/spack/cache
 
 
   # Cache directory for miscellaneous files, like the package index.
-  # This can be purged with `spack purge --misc-cache`
-  misc_cache: .spack/misccache
+  # This can be purged with `spack clean --misc-cache`
+  misc_cache: ~/.spack/cache
```

I have recently ran into issues with environments where the misc_cache needed to be cleaned because I was switching to an older spack.
In order to keep Uberenv "bullet-proof" I think the misc_cache should remain uberenv-local, e.g.
```suggestion
  misc_cache: $spack/.misc_cache
```

---

Good catch, I think I may have missed this when updating the `config.yaml`